src/test/platform/interpreter/common/variables/customEnvironmentVariablesProvider.node.vscode.test.ts

```diff
@@ -0,0 +1,231 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+import { assert } from 'chai';
+import { Uri } from 'vscode';
+import { IWorkspaceService } from '../../../../../platform/common/application/types';
+import { disposeAllDisposables } from '../../../../../platform/common/helpers';
+import { IDisposable } from '../../../../../platform/common/types';
+import { isWeb } from '../../../../../platform/common/utils/misc';
+import { CustomEnvironmentVariablesProvider } from '../../../../../platform/common/variables/customEnvironmentVariablesProvider.node';
+import { IEnvironmentVariablesService } from '../../../../../platform/common/variables/types';
+import { IS_REMOTE_NATIVE_TEST } from '../../../../constants';
+import { initialize } from '../../../../initialize';
+import * as fs from 'fs-extra';
+import { EXTENSION_ROOT_DIR_FOR_TESTS } from '../../../../constants.node';
+import * as dedent from 'dedent';
+import { IPythonExtensionChecker } from '../../../../../platform/api/types';
+import { createEventHandler } from '../../../../common';
+
+suite('Custom Environment Variables Provider', () => {
```

Comment:
could these be unit tests? Doesn't seem to requre vscode to be running

Reply:
I guess this all depends on the definition of a unit test, should evertying be mocked? If yes, then thats not a good test.

I could mock a lot of things in the test, like reading and parsing the .env file and creating and editing files and then emitting file change evengts, etc... thats a heck of a lot to mock, and the effort to write the test using the real bits is significantly simpler and gets the job done.

basically I want to the the outputs for given inputs & that's exactly what this test it doing
& I rely on vscode here as I'm using the VS Code API to create the file watcher and i don't want to mock that, too much work and not worth it at all.

Here we have 6 test and everything runs in <1s.

Reply:
@amunger looks like i'll have to create unit tests and mock the vscode classes, as the file change detection doesn't work on CI, possibly due to a large number of files or other.